src/sema/expression.rs

```diff
@@ -5852,6 +5867,11 @@ fn method_call_pos_args(
                         symtable,
                         diagnostics,
                     );
+                } else {
+                    diagnostics.push(Diagnostic::error(
+                        *loc,
+                        "function calls via contract name are only valid for base contracts".into(),
+                    ));
```

Comment:
The same change is needed in `fn method_call_named_args()` and this needs a test case too.

```solidity
contract B {
    function b() public {
        Base.set({val: 1});
    }
}
```